src/components/CommentAdder.jsx

```diff
@@ -0,0 +1,56 @@
+import { useState, useContext } from "react";
+import { useParams } from "react-router-dom";
+import { UserContext } from "../context/UserContext";
+import { postComment } from "../utils/api";
+
+const PostComment = ({ setComments }) => {
+  const { article_id } = useParams();
+  const [newComment, setNewComment] = useState("");
+  const { user } = useContext(UserContext);
+  const [error, setError] = useState(null);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    postComment(article_id, newComment, user.username)
+      .then((postedComment) => {
+        setComments((currComments) => {
+          return [postedComment, ...currComments];
+        });
+      })
+      .catch((error) => {
+        setError(error.response.data);
+      });
+    setNewComment("");
+  };
+
+  const handleInput = (event) => {
+    const { value } = event.target;
+    setNewComment(value);
+  };
+
+  if (error) {
+    return <p> Please try again!</p>;
+  }
+
+  if (user.username !== "") {
+    return (
+      <>
+        <form onSubmit={handleSubmit}>
+          <label>Add a comment</label>
+          <input
+            type="text"
+            name="body"
+            value={newComment}
+            onChange={handleInput}
```

Comment:
You may want to add a prevent default on your handleInput - if I press return the input box clears 
You could also consider looking at textarea html tag instead of an input box - slightly better UX

Reply:
Will work on this